File: LeetCode/0202. Happy Number/JavaScript/01. Happy Number.js
var isHappy = function (n) {
    let intSets = new Set();
    while (n != 1) {
        let newPowSum = 0;
        Array.from(n.toString()).forEach(function (k) {
            newPowSum += Math.pow(parseInt(k), 2);
        });
        if (intSets.has(newPowSum))
            return false;
        intSets.add(newPowSum);
        n = newPowSum;
    }
    return true;
};

File: LeetCode/0202. Happy Number/C#/01. Happy Number.cs
bool IsHappy(int n)
{
    HashSet<int> intSets = new HashSet<int>();
    while (n != 1)
    {
        int newPowSum = 0;
        foreach (char k in n.ToString().ToCharArray())
            newPowSum += (int)Math.Pow(int.Parse(k.ToString()), 2);
        if (intSets.Contains(newPowSum))
            return false;
        intSets.Add(newPowSum);
        n = newPowSum;
    }
    return true;
}